functional_tests/fixtures.py

```diff
@@ -58,16 +59,24 @@ def detected_panics(zip_file_path):
 
 def detected_pattern(pattern, zip_file_path):
 
-    if not os.path.isfile(zip_file_path):
-        log.error("Can't run zipgrep, cannot find zipfile: {}".format(zip_file_path))
+    if not zip_file_path:
         return False
+    
+    try:
+        if not os.path.isfile(zip_file_path):
+            log.error("Can't run zipgrep, cannot find zipfile: {}".format(zip_file_path))
+            return False
+
+        process = Popen(["zipgrep", pattern, zip_file_path], stdout=PIPE)
+        (output, err) = process.communicate()
+        exit_code = process.wait()
+        if exit_code == 0:
```

Comment:
I don't think that will work because it won't capture the output, which is needed here: `log.info("Detected pattern {}: {}".format(pattern, output))`